src/jlfrontend.scm

```diff
@@ -79,11 +67,22 @@
     (if (and (pair? ex0) (eq? (car ex0) 'toplevel))
         ex0
         (let* ((ex (julia-expand0 ex0))
-               (gv (toplevel-expr-globals ex))
+               (lv (find-decls 'local ex))
+               (gv (diff (delete-duplicates
+                           (append (find-decls 'const ex) ;; convert vars declared const outside any scope block to outer-globals
+                                   (find-decls 'global ex) ;; convert vars declared global outside any scope block to outer-globals
+                                   ;; vars assigned at the outer level
+                                   (filter (lambda (x) (not (some-gensym? x)))
+                                           (find-assigned-vars ex '() '()))))
+                         lv))
+               ;; vars assigned anywhere, if they have been been explicitly defined
+               (gv-implicit (filter (lambda (x) (and (not (or (memq x lv) (memq x gv))) (defined-julia-global x)))
```

Comment:
I thought this change would mean lowering doesn't depend on `defined-julia-global`?